### PR TITLE
[master] Deprecate `pushover` modules in favor of saltext

### LINKED ISSUE #65567: [TECH DEBT] Deprecate Pushover modules
The `pushover` modules will be migrated to https://github.com/salt-extensions/saltext-pushover in 3009.0.

Files to remove:

```
salt/modules/pushover_notify.py
salt/states/pushover.py
salt/returners/pushover_returner.py
doc/ref/modules/all/salt.modules.pushover_notify.rst
doc/ref/states/all/salt.states.pushover.rst
doc/ref/returners/all/salt.returners.pushover_returner.rst
```

The modules do not have any tests.

### PR DESCRIPTION
**Pull request by lkubb**
### What does this PR do?
Deprecates the `pushover` execution/state/returner module in favor of the Salt Extension found at https://github.com/salt-extensions/saltext-pushover. Removal target is 3009.0.

### What issues does this PR fix or reference?
Fixes: https://github.com/saltstack/salt/issues/65567

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [x] Docs
- [x] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [ ] Tests written/updated

### Commits signed with GPG?
Yes

### CODE REVIEW

File: salt/modules/pushover_notify.py [hunk 1]
```diff
@@ -1,6 +1,11 @@
 """
 Module for sending messages to Pushover (https://www.pushover.net)
 
+.. warning::
+    This module will be removed from Salt in version 3009.0 in favor of
+    the `saltext.pushover Salt Extension
+    <https://github.com/salt-extensions/saltext-pushover>`_
+
```
Comment (s0undt3ch) [inline salt/modules/pushover_notify.py:8]:
```suggestion
```
This change made me think we should automate these warnings. Thank You!

https://github.com/saltstack/salt/pull/65572

File: salt/returners/pushover_returner.py
Comment (s0undt3ch) [inline salt/returners/pushover_returner.py:1]: This module does not have the `__deprecated__` attribute because?

File: salt/states/pushover.py [hunk 2]
```diff
@@ -2,6 +2,11 @@
 Send a message to PushOver
 ==========================
 
+.. warning::
+    This module will be removed from Salt in version 3009.0 in favor of
+    the `saltext.pushover Salt Extension
+    <https://github.com/salt-extensions/saltext-pushover>`_
+
```
Comment (s0undt3ch) [inline salt/states/pushover.py:9]:
```suggestion
```

**Review by s0undt3ch — CHANGES_REQUESTED** [review]

File: salt/returners/pushover_returner.py
Comment (lkubb) [reply salt/returners/pushover_returner.py:1]: ... I must have forgotten to save the buffer. Sorry!

**Review by s0undt3ch — APPROVED** [review]

**Review by twangboy — APPROVED** [review]

**Pull request merged**